browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -68,44 +68,37 @@ void SetAdblockCnameHostResolverForTesting(
   g_testing_host_resolver = host_resolver;
 }
 
-void ShouldBlockAdOnTaskRunner(std::shared_ptr<BraveRequestInfo> ctx,
-                               base::Optional<std::string> canonical_name) {
-  bool did_match_rule = false;
-  bool did_match_exception = false;
-  bool did_match_important = false;
+// If `canonical_url` is specified, this will only check if the CNAME-uncloaked
+// response should be blocked. Otherwise, it will run the check for the
+// original request URL.
+void ShouldBlockRequestOnTaskRunner(std::shared_ptr<BraveRequestInfo> ctx,
+                                    base::Optional<GURL> canonical_url) {
   if (!ctx->initiator_url.is_valid()) {
     return;
   }
   std::string source_host = ctx->initiator_url.host();
 
-  g_brave_browser_process->ad_block_service()->ShouldStartRequest(
-      ctx->request_url, ctx->resource_type, source_host, &did_match_rule,
-      &did_match_exception, &did_match_important, &ctx->mock_data_url);
-  if (did_match_important) {
-    ctx->blocked_by = kAdBlocked;
-    return;
+  GURL url_to_check;
+  if (canonical_url.has_value()) {
+    url_to_check = *canonical_url;
+  } else {
+    url_to_check = ctx->request_url;
   }
 
-  if (canonical_name.has_value() &&
-      ctx->request_url.host() != *canonical_name && *canonical_name != "") {
-    GURL::Replacements replacements = GURL::Replacements();
-    replacements.SetHost(
-        canonical_name->c_str(),
-        url::Component(0, static_cast<int>(canonical_name->length())));
-    const GURL canonical_url = ctx->request_url.ReplaceComponents(replacements);
-
-    g_brave_browser_process->ad_block_service()->ShouldStartRequest(
-        canonical_url, ctx->resource_type, source_host, &did_match_rule,
-        &did_match_exception, &did_match_important, &ctx->mock_data_url);
-  }
+  g_brave_browser_process->ad_block_service()->ShouldStartRequest(
+      url_to_check, ctx->resource_type, source_host, &ctx->did_match_rule,
+      &ctx->did_match_exception, &ctx->did_match_important,
+      &ctx->mock_data_url);
 
-  if (did_match_important || (did_match_rule && !did_match_exception)) {
+  if (ctx->did_match_important ||
+      (ctx->did_match_rule && !ctx->did_match_exception)) {
     ctx->blocked_by = kAdBlocked;
   }
 }
 
-void OnShouldBlockAdResult(const ResponseCallback& next_callback,
-                           std::shared_ptr<BraveRequestInfo> ctx) {
+void OnShouldBlockUncloakedRequestResult(
```

Comment:
we probably can merge it with `OnShouldBlockRequestResult` ?

Reply:
They do look similar but I don't really see a good way to do that - in `OnShouldBlockUncloakedRequestResult` we need to always run the next callback, but in `OnShouldBlockRequestResult` we only run it when the request is blocked and check the CNAME otherwise.

Reply:
well i' would just add a boolean and bind it. Looks better than copying the `Dispatch` call

Reply:
fixed